server/repository/src/db_diesel/number_row.rs

```diff
@@ -40,6 +46,32 @@ impl fmt::Display for NumberRowType {
             NumberRowType::RequestRequisition => write!(f, "REQUEST_REQUISITION"),
             NumberRowType::ResponseRequisition => write!(f, "RESPONSE_REQUISITION"),
             NumberRowType::Stocktake => write!(f, "STOCKTAKE"),
+            NumberRowType::Program(custom_string) => write!(f, "PROGRAM_{}", custom_string),
+        }
+    }
+}
+
+impl TryFrom<String> for NumberRowType {
+    type Error = NumberRowTypeError;
+
+    fn try_from(s: String) -> Result<Self, NumberRowTypeError> {
```

Comment:
Is try_from actually used for anything else than testing? do we need it? NumberRowTypeError is also not used?

Reply:
Just testing I think at this stage. I guess theoretically it shouldn't ever be needed, the Service API is always about finding the next value for Number Row type. Shouldn't be any reason to map back the other way except for maybe in sync or testing?

Reply:
No I don't think it is needed

Reply:
sorry its needed for the sync translation though